x-pack/plugins/security/server/authentication/providers/saml.test.ts

```diff
@@ -325,6 +353,177 @@ describe('SAMLAuthenticationProvider', () => {
+        expect(authenticationResult.error).toEqual(
+          Boom.badRequest('State does not include URL path to redirect to.')
+        );
+      });
+
+      it('does not handle AJAX requests.', async () => {
+        const request = httpServerMock.createKibanaRequest({ headers: { 'kbn-xsrf': 'xsrf' } });
+
+        const authenticationResult = await provider.login(
+          request,
+          {
+            step: SAMLLoginStep.RedirectURLFragmentCaptured,
+            redirectURLFragment: '#some-fragment',
+          },
+          { redirectURL: '/test-base-path/some-path' }
+        );
+
+        sinon.assert.notCalled(mockOptions.client.callAsInternalUser);
+
+        expect(authenticationResult.notHandled()).toBe(true);
+      });
+
+      it('redirects non-AJAX requests to the IdP remembering combined redirect URL.', async () => {
+        const request = httpServerMock.createKibanaRequest();
+
+        mockOptions.client.callAsInternalUser.withArgs('shield.samlPrepare').resolves({
+          id: 'some-request-id',
+          redirect: 'https://idp-host/path/login?SAMLRequest=some%20request%20',
+        });
+
+        const authenticationResult = await provider.login(
+          request,
+          {
+            step: SAMLLoginStep.RedirectURLFragmentCaptured,
+            redirectURLFragment: '#some-fragment',
+          },
+          { redirectURL: '/test-base-path/some-path' }
+        );
+
+        sinon.assert.calledWithExactly(
+          mockOptions.client.callAsInternalUser,
+          'shield.samlPrepare',
+          { body: { realm: 'test-realm' } }
+        );
+
+        expect(mockOptions.logger.warn).not.toHaveBeenCalled();
+
+        expect(authenticationResult.redirected()).toBe(true);
+        expect(authenticationResult.redirectURL).toBe(
+          'https://idp-host/path/login?SAMLRequest=some%20request%20'
+        );
+        expect(authenticationResult.state).toEqual({
+          requestId: 'some-request-id',
+          redirectURL: '/test-base-path/some-path#some-fragment',
+        });
+      });
+
+      it('prepends redirect URL fragment with `#` if it does not have one.', async () => {
+        const request = httpServerMock.createKibanaRequest();
+
+        mockOptions.client.callAsInternalUser.withArgs('shield.samlPrepare').resolves({
+          id: 'some-request-id',
+          redirect: 'https://idp-host/path/login?SAMLRequest=some%20request%20',
+        });
+
+        const authenticationResult = await provider.login(
+          request,
+          {
+            step: SAMLLoginStep.RedirectURLFragmentCaptured,
+            redirectURLFragment: '../some-fragment',
+          },
+          { redirectURL: '/test-base-path/some-path' }
+        );
+
+        sinon.assert.calledWithExactly(
+          mockOptions.client.callAsInternalUser,
+          'shield.samlPrepare',
+          { body: { realm: 'test-realm' } }
+        );
+
+        expect(mockOptions.logger.warn).toHaveBeenCalledTimes(1);
```

Comment:
__note:__ I know and agree in advance that testing things like this (`logger.warn`) feels redundant, but it doesn't hurt and I can't think of any occasion when I suffered from anything like this during whole auth provider code life.

Reply:
Ensuring a warn is called seems reasonable here to me!